algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala

```diff
@@ -329,31 +329,35 @@ case class SparseHLL(bits: Int, maxRhow: Map[Int, Max[Byte]]) extends HLL {
         if (allMaxRhow.size * 16 <= size) {
           SparseHLL(bits, allMaxRhow)
         } else {
-          DenseHLL(bits, sparseMapToSequence(allMaxRhow))
+          DenseHLL(bits, sparseMapToArray(allMaxRhow))
         }
 
       case DenseHLL(bits, oldV) =>
         assert(oldV.size == size, "Incompatible HLL size: " + oldV.size + " != " + size)
-        val newV = maxRhow.foldLeft(oldV) {
-          case (v, (j, rhow)) =>
-            if (rhow.get > v(j)) {
-              v.updated(j, rhow.get)
-            } else {
-              v
-            }
+        val newContents: Array[Byte] = oldV.array.clone
+        val siz = newContents.size
+
+        maxRhow.foreach {
```

Comment:
should we while loop it here to be consistent? We could get the iterator, and do the hasNext, next unrolling?

Man, we really need some macros for this stuff (glanced at ScalaBlitz, does not seem to be exactly what we want here).